test/e2e/e2e_setup_test.go

```diff
@@ -73,6 +74,13 @@ func WithPrefix(variableName string) SetupOption {
 	}
 }
 
+// WithUseKindForManagementCluster instructs Setup to run extra steps for the separate kind management cluster.
+func WithUseKindForManagementCluster() SetupOption {
```

Comment:
might be: withAdditionalVCSimServer(bool)

and the bool is a var that we assign as a input to this function and as a value for ClusterctlUpgradeSpecInput.UseKindForManagementCluster, so it surfaces the correlation between the two settings

Reply:
I find this very confusing to be honest